src/main/java/org/broadinstitute/dsm/DSMServer.java

```diff
@@ -295,9 +304,56 @@ protected void setupCustomRouting(@NonNull Config cfg) {
 
         setupJobs(cfg, kitUtil, notificationUtil, eventUtil, container, receiver);
 
+        setupPubSub(cfg);
+
         logger.info("Finished setting up DSM custom routes and jobs...");
     }
 
+    private void setupPubSub(@NonNull Config cfg) {
+        String projectId = cfg.getString(GCP_PATH_TO_PUBSUB_PROJECT_ID);
+        String subscriptionId = cfg.getString(GCP_PATH_TO_PUBSUB_SUB);
+
+        try {
+            ProjectSubscriptionName subscriptionName =
+                    ProjectSubscriptionName.of(projectId, subscriptionId);
+
+            // Instantiate an asynchronous message receiver.
+            MessageReceiver receiver =
+                    (PubsubMessage message, AckReplyConsumer consumer) -> {
+                        // Handle incoming message, then ack the received message.
+                        try {
+                            PubSubLookUp.processCovidTestResults(message);
+                            logger.info("Processing the message finished");
+                            consumer.ack();
+
+                        }catch(Exception ex){
+                            logger.info("about to nack the message");
```

Comment:
It's better to log the exception instead of printing stacktrace directly.
```suggestion
                            logger.info("about to nack the message", ex);
```